src/visualization/components/internal/AdaptiveZoomOption.tsx

```diff
@@ -0,0 +1,72 @@
+// Libraries
+import React, {CSSProperties, FC} from 'react'
+
+// Components
+import {
+  AlignItems,
+  ComponentSize,
+  FlexBox,
+  FlexDirection,
+  InfluxColors,
+  InputLabel,
+  QuestionMarkTooltip,
+  SlideToggle,
+} from '@influxdata/clockface'
+
+// Metrics
+import {event} from 'src/cloud/utils/reporting'
+
+interface AdaptiveZoomToggleProps {
+  adaptiveZoomHide: boolean
+  update: (obj: any) => void
+  testID?: string
+  type: string
+}
+
+const getToggleColor = (toggle: boolean): CSSProperties => {
+  if (toggle) {
+    return {color: InfluxColors.Grey95}
+  }
+  return {color: InfluxColors.Grey65}
+}
+
+export const AdaptiveZoomToggle: FC<AdaptiveZoomToggleProps> = ({
+  adaptiveZoomHide,
+  update,
+  testID = 'adaptive-zoom-toggle',
+  type,
+}) => {
+  const handleSetAdaptiveZoom = () => {
+    update({
+      adaptiveZoomHide: !adaptiveZoomHide,
+    })
+    event(`visualization.customize.adaptiveZoom.${!adaptiveZoomHide}`, {
```

Comment:
Does using `!adaptiveZoomHide` invert the meaning of what happened? Or does it describe what the value is changing to?

We generally want to move away from variable event names, but since this is a boolean value, it's not a huge deal. One stylistic change I'd suggestion is consider making it more human readable, by saying something like `on` or `off` instead of `true` or `false`, so it's more clear to people analazying the data what happened

```ts
event(`visualization.customize.adaptiveZoom.${adaptiveZoomHide ? 'toggleOn' : 'toggleOff'}`
```

or something to that effect.

Reply:
`adaptiveZoomHide` is the current setting. The toggle flips the current setting to the other setting. So, `!adaptiveZoomHide` would be the opposite of the current setting, or flipping the current setting.

Reply:
i would argue that 

```
event(`visualization.customize.adaptiveZoom.${adaptiveZoomHide ? 'toggleOn' : 'toggleOff'}
```

is more confusing that necessary. Because we are not firing an event on the current state of `adaptiveZoomHide`. We are firing an event when the user flips the toggle.